Core/src/main/java/com/rakuten/tech/mobile/perf/core/base/WebViewClientBase.java

```diff
@@ -20,9 +23,19 @@ public void onPageStarted(WebView view, String url, Bitmap favicon) {
   public void onPageFinished(WebView view, String url) {
     Tracker.prolongMetric();
     if (com_rakuten_tech_mobile_perf_page_trackingId != 0) {
-      Tracker.endUrl(com_rakuten_tech_mobile_perf_page_trackingId);
+      Tracker.endUrl(com_rakuten_tech_mobile_perf_page_trackingId, 200);
       com_rakuten_tech_mobile_perf_page_trackingId = 0;
     }
     super.onPageFinished(view, url);
   }
+
+  public void onReceivedHttpError(WebView view, WebResourceRequest request,
```

Comment:
this method was added in API 23 - so if we run this on lower API levels the super call & the lack of WebResourceResponse might cause problems, won't it? Have you tested for that?
If so there are Annotations in place for methods that are only available on certain API levels.

Alternatively you could also use WebViewClient#onReceivedError which is available since API level 1 (and deprecated now).
See https://developer.android.com/reference/android/webkit/WebViewClient.html#onReceivedError(android.webkit.WebView, int, java.lang.String, java.lang.String)

Reply:
I have checked it on API 15 & API 16 emulator, that WebViewClient#onReceivedHttpError is not been called.

For WebViewClient#onReceivedError, will only be called when you non http error(like network cut, timeout etc). It's not within this sprint's scope.

Reply:
ok, let's have QA confirm this for all available API levels just to be sure